### PR TITLE
Add ability to add/remove vote for the resource - fixes #291

### LINKED ISSUE #291: Upvoting resources on a skill
Implement upvoting resources on a skill

References https://github.com/x-team/unleash/pull/287
## Requirements
- [ ] #293

### PR DESCRIPTION
**Pull request by nicksp**
One vote per user is allowed both by backend and frontend. Once voted, the thumb button is disabled.

### CODE REVIEW

**Comment by nicksp** [comment]
@jacekelgda @karolsojko I guess this is ready for review.

File: app/reducers/skillsReducer.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
  * @param  {object}  skill      The skill to update.
  * @return {object}             The updated result.
  */
 function updateOne(skills, skill) {
-  const upsdatedSkills = cloneDeep(skills);
-  upsdatedSkills[skill.name] = skill;
-  return upsdatedSkills;
+  const updatedSkills = cloneDeep(skills);
+  updatedSkills[skill.name] = skill;
+  return updatedSkills;
```
Comment (szymonmichalak) [inline app/reducers/skillsReducer.js]: This won't work. Skills are not indexed by `name`, `skills` is an array of objects. I would suggest to change that to something like this:

```
function updateOne(skills, skill) {
  const indexOfSkill = findIndex(skills, { id: skill.id });
  return [
    ...skills.slice(0, indexOfSkill),
    skill,
    ...skills.slice(indexOfSkill + 1),
  ];
}
```

You can also use `cloneDeep` and `updateSkills[indexOfSkill]`, I just like above-mentioned approach ;-)

Comment (nicksp) [inline app/reducers/skillsReducer.js]: Ahh i see now ;) You're right! Will definitely change the update logic.
Thanks for your suggestion @szymonmichalak

**Comment by nicksp** [comment]
@szymonmichalak Did some changes as per your suggestion. Ready for re-review :)

**Comment by nicksp** [comment]
There is actually a strange behaviour, when you downvote API returns incorrect `votes_total`. But that might be fixed in another repo and it will work with my changes properly then.

File: app/components/Skill.jsx [hunk 2]
```diff
@@ -240,13 +264,17 @@ class Skill extends Component {
...
     const skill = this.getSkillBySlug(skills, this.props.params.slug);
     const skilled = this.getProfilesInIds(profiles, profilesBySkill);
-    const resources = get(skill, 'resources', []).map(resource => ({
-      id: resource.id,
-      url: resource.url,
-      upvotes: random(0, 10),
-      upvoted: random(0, 3) === 0,
-      type: resourceTypes[resource.type] ? resource.type : 'other',
-    })).sort((a, b) => b.upvotes - a.upvotes);
+    const resources = get(skill, 'resources', []).map((resource) => {
```
Comment (karolsojko) [inline app/components/Skill.jsx]: maybe it's a good idea to encapsulate this whole mapping of resources from a skill (and sorting) into a method. What do you think @szymonmichalak @kdemoya  - maybe there's potential for a helper?

**Comment by nicksp** [comment]
After some slack discussion with @szymonmichalak we decided to go with the following logic for resources:

1. We always show 👍 icon, cause it might feel strange to have smth. like this 👎 x 2
1. Change colour upon vote action: black (default initial colour) - is downvoted/not clicked resource, pink - upvoted one.
1. The loading spinner we decided to keep for another PR as it's smth. that will require some investigation first.

File: app/components/Skill.jsx (hunk 2 above)
Comment (szymonmichalak) [inline app/components/Skill.jsx]: 👍 sure, definitely worth doing

Comment (nicksp) [inline app/components/Skill.jsx]: @karolsojko @szymonmichalak  Done :)

**Comment by jacekelgda** [comment]
So, can we merge this if 2 reviewers approved ?

**Comment by szymonmichalak** [comment]
Fixes #291

**Comment by nicksp** [comment]
@jacekelgda @szymonmichalak Why wasn't it  triunleashed ;) ?

**Pull request merged**